internal/log_analysis/log_processor/logtypes/entry.go

```
@@ -0,0 +1,279 @@
+		newParser:     fac.NewParser,
+		glueTableMeta: awsglue.NewGlueTableMetadata(models.LogData, desc.Name, desc.Description, awsglue.GlueTableHourly, schema),
+	}
+}
+
+func (e *entry) Name() string {
+	return e.desc.Name
+}
+
+func (e *entry) Find(logType string) Entry {
+	if e.String() == logType {
+		return e
+	}
+	return nil
+}
+
+func (e *entry) Len() int {
+	return 1
+}
+
+func (e *entry) Entries() []Entry {
+	return []Entry{e}
+}
+
+func (e *entry) BuildEntry() (Entry, error) {
+	return e, nil
+}
+
+func (e *entry) Describe() Desc {
+	return e.desc
+}
+
+func (e *entry) String() string {
+	return e.desc.Name
+}
+
+func (e *entry) Schema() interface{} {
+	return e.schema
+}
+
+// GlueTableMeta returns the glue table metadata for this entry
+func (e *entry) GlueTableMeta() *awsglue.GlueTableMetadata {
+	return e.glueTableMeta
+}
+
+// Parser returns a new parsers.Interface instance for this log type
+func (e *entry) NewParser(params interface{}) (parsers.Interface, error) {
+	return e.newParser(params)
+}
+
+func checkLogEntrySchema(logType string, schema interface{}) error {
+	if schema == nil {
+		return errors.Errorf("nil schema for log type %q", logType)
+	}
+	data, err := jsoniter.Marshal(schema)
+	if err != nil {
+		return errors.Errorf("invalid schema struct for log type %q: %s", logType, err)
+	}
+	var fields map[string]interface{}
+	if err := jsoniter.Unmarshal(data, &fields); err != nil {
+		return errors.Errorf("invalid schema struct for log type %q: %s", logType, err)
+	}
+	// Verify we can generate glue schema from the provided struct
+	if err := checkGlue(schema); err != nil {
+		return errors.Wrapf(err, "failed to infer Glue columns for %q", logType)
+	}
+	return nil
+}
+
+func checkGlue(schema interface{}) (err error) {
+	defer func() {
+		if e := recover(); e != nil {
+			switch e := e.(type) {
+			case error:
+				err = e
+			case string:
+				err = errors.New(e)
+			default:
+				err = errors.Errorf(`panic %v`, e)
+			}
+		}
+	}()
```

Why do we need this handling here?

---

Because glue infer code panics with different types of error instead of returning errors. I don't know if this has been streamlined since I wrote it. Note that this is not new code, it just moved to a different file.